wallets/exodus-extension/src/extension/client.ts

```diff
@@ -13,37 +15,42 @@ export class ExodusClient implements WalletClient {
     this.client = client;
   }
 
+  async connect(chainId: string | string[]) {
+    await this.client.connect({
+      chainId: Array.isArray(chainId) ? chainId[0] : chainId,
+    });
+  }
+
   async getAccount(chainId: string) {
-    const account = await this.client.connect({ chainId });
+    const response = await this.client.connect({ chainId });
 
-    this.account = account;
+    this.account = {
+      ...response,
+      pubkey: response.publicKey,
+    };
 
-    return account;
+    return this.account;
   }
 
-  async getOfflineSigner() {
+  async getOfflineSigner(chainId: string) {
+    await this.getAccount(chainId);
```

Comment:
Just curious if a dapp expects that `getOfflineSigner` initiates a connection flow. Asking because `getAccounts` alternatively can be implemented like:
```
getAccounts: async (): Promise<AccountData[]> => this.getAccount(chainId)
```

Reply:
not sure but the reason for calling the method (`getAccount`) there is to make sure `this.account` has been set before. Will update it to use your suggested implementation for triggering the method only when needed

Reply:
Sounds good, thanks!